### ISSUE TITLE
`UsersIndex.create!` not working with ES version 7.x

### ISSUE THREAD

**Issue by sachin1mittal** [toptal/chewy#708]
I have a users_index class as following: 

`app/indexes/users_index.rb`

```
class UsersIndex < Chewy::Index
  define_type User.active do
    field :first_name
  end
end

```
With ES version 7.6.2, When I run `UsersIndex.create!`, it raises an error `"type": "mapper_parsing_exception",
        "reason": "Root mapping definition has unsupported parameters:` 
While when I do the same with version 6.8.8, it works fine.

I read that the body argument structure has been changed in ES version 7.

So, with version 7, request format should be:

```
PUT /users
{
  "mappings": {
      "properties": {
        "first_name": {
          "type": "text"
        }
    }
  }
}
```

While for earlier versions, it should be

```
PUT /users
{
  "mappings": {
     "user": {
        "properties": {
          "first_name": {
            "type": "text"
          }
      }
    }
  }
}
```

And, its failing because of this change.

I think changing `chewy-5.1.0/lib/chewy/fields/root.rb#33` from `mapping` to `mapping[name]` should make it work fine.

Basically, it's returning the mapping hash as 

```
{
    "user": {
        "properties": {
            "first_name": {
                "type": "text"
            }
        }
    }
}
```
While, for the 7.x versions, it should return as

```
 {
    "properties": {
        "first_name": {
            "type": "text"
        }
    }
}
```

**Comment by przbadu** [comment]
I am facing same issue, after searching for a while, ES 7.x has removed mapping types https://www.elastic.co/guide/en/elasticsearch/reference/current/removal-of-types.html

**Comment by przbadu** [comment]
@sachin1mittal is correct, returning `mapping[name]` is working and indexing mapping correctly for me.

**Comment by silva96** [comment]
related to #673

**Comment by reneklacan** [comment]
@sachalifs before it's fixed, you can try this nasty monkeypatch which fixes it for us on ES 7.8: https://gist.github.com/reneklacan/5ed4bd08b0ad43cd81cab38530ae0889

**Comment by AlexWayfer** [comment]
> @sachalifs before it's fixed, you can try this nasty monkeypatch which fixes it for us on ES 7.8: https://gist.github.com/reneklacan/5ed4bd08b0ad43cd81cab38530ae0889

Maybe you want to create a PR with these fixes?

**Comment by rabotyaga** [comment]
Please take a look at [Chewy 7.0.0 release](https://github.com/toptal/chewy/releases/tag/v7.0.0). Stay tuned for the 7.1 release, with dropped support for ES6.8 and removed deprecation warnings.
